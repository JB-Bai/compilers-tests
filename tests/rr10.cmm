float wHMCpGWnW(int _66iWf5i){
	int L7GMi = AotcIzMFrR;
	832.11;
	return E2AOiv7J() !=(932.411E-1 = T2mSnog + 463.882 * 8615 / VadFv(! 6819 + Emtkz || 426 = pHtSpYAR ||(_l1MwMTXXQQ && 00) = ! !(3621.ig4Gi846) == tJ0T6p == 050.22E2 / hmYX5Yce && 064415 > 170953 && 0x88fEd || 279.8 || 01405 * 834.276E-0.sUpsPeOdN * 130.13E0 /(h9FmcczyZf * 045305.OB5F2.MqXTklUkdkm) || R1yWal || - 301.1022 = XRSSlugn(Ot9cOjKTA, 0x43) && 0x0 || - 0x856 && 0xbd || ! 01653331 / uDn(6527.193e+7 || 79.86, 2.235 * eofrN4() / a1_ZYF1eXu > TjhP_na3() > iIkFME0() || 06614.M8yRScvm = 0167110 + 0x518 <= ! IolK5X() / pHIponW * 0.788), - 0x3d3 || ! 014356[- ! ! ! OBMfYSX() < sg6yTxmB && ! 02206 < O6RzfjwHp() + npUWxdGj0]) && 4.9645 / 8.9 - 2491);
	! 3848.358E-8 && EDczaDe;
	if(Ysyy6vB || 05017040 <= O39q3JpQz(04310 * 04040).JJoPm2 + 38.9E+1) 64.158E-1 - eMg1t8;
}
struct{
	int La7LG2_6_Aw[0xF5fd], I4vL87mj3 = 622277 = JbkjiC4b(ylpyb_(0x6e)), VXJeN, m5y1Q1Y9 = 40.28;
	float SQEup9[0573][0456], Qtixt0HLG3_;
	float dznEm[9237][02014];
	int sKezEt = k8TeL3(iO2Jnx(), QI6_M_, 1400.4147E-6, IPnDn3dle[DSEZ4Ie] ==(51.79e0[EcMYvqls || ! 80988 / 75968.aI0lfWN.GuuKpZH]), ! 5.342E5,(01.09E+1 = u2kUMN) / 3165.6e+2 || ! - ikfM(W70ir / jDPylQF(919.27 * hm9Ulz(XdxGuieEPu.enPvJK, 4130.309e8 + PVna)), 12.312E+6, YkiveSwM, 3780.93E2) - ! 0x6af * 401.664e1 * 0267 * 1.7 == 02014.kPQDdsg && 4.541) && 7245.32 + 01176;
}
;
float SeNi2Z, V3pnGzkQEKr[11][0x1E9], U4kFnX5QmE[050316], FDV8umb[9806], Z8Tbar, Skm6YH, l2yCIRu;

