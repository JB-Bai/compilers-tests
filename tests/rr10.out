Program (1)
  ExtDefList (1)
    ExtDef (1)
      Specifier (1)
        TYPE: float
      FunDec (1)
        ID: wHMCpGWnW
        LP
        VarList (1)
          ParamDec (1)
            Specifier (1)
              TYPE: int
            VarDec (1)
              ID: _66iWf5i
        RP
      CompSt (1)
        LC
        DefList (2)
          Def (2)
            Specifier (2)
              TYPE: int
            DecList (2)
              Dec (2)
                VarDec (2)
                  ID: L7GMi
                ASSIGNOP
                Exp (2)
                  ID: AotcIzMFrR
            SEMI
        StmtList (3)
          Stmt (3)
            Exp (3)
              FLOAT: 832.109985
            SEMI
          StmtList (4)
            Stmt (4)
              RETURN
              Exp (4)
                Exp (4)
                  ID: E2AOiv7J
                  LP
                  RP
                RELOP
                Exp (4)
                  LP
                  Exp (4)
                    Exp (4)
                      FLOAT: 93.241096
                    ASSIGNOP
                    Exp (4)
                      Exp (4)
                        Exp (4)
                          ID: T2mSnog
                        PLUS
                        Exp (4)
                          Exp (4)
                            Exp (4)
                              FLOAT: 463.881989
                            STAR
                            Exp (4)
                              INT: 8615
                          DIV
                          Exp (4)
                            ID: VadFv
                            LP
                            Args (4)
                              Exp (4)
                                Exp (4)
                                  Exp (4)
                                    Exp (4)
                                      NOT
                                      Exp (4)
                                        INT: 6819
                                    PLUS
                                    Exp (4)
                                      ID: Emtkz
                                  OR
                                  Exp (4)
                                    INT: 426
                                ASSIGNOP
                                Exp (4)
                                  Exp (4)
                                    Exp (4)
                                      ID: pHtSpYAR
                                    OR
                                    Exp (4)
                                      LP
                                      Exp (4)
                                        Exp (4)
                                          ID: _l1MwMTXXQQ
                                        AND
                                        Exp (4)
                                          INT: 0
                                      RP
                                  ASSIGNOP
                                  Exp (4)
                                    Exp (4)
                                      Exp (4)
                                        Exp (4)
                                          Exp (4)
                                            Exp (4)
                                              Exp (4)
                                                Exp (4)
                                                  Exp (4)
                                                    Exp (4)
                                                      NOT
                                                      Exp (4)
                                                        NOT
                                                        Exp (4)
                                                          LP
                                                          Exp (4)
                                                            Exp (4)
                                                              INT: 3621
                                                            DOT
                                                            ID: ig4Gi846
                                                          RP
                                                    RELOP
                                                    Exp (4)
                                                      ID: tJ0T6p
                                                  RELOP
                                                  Exp (4)
                                                    Exp (4)
                                                      FLOAT: 5022.000000
                                                    DIV
                                                    Exp (4)
                                                      ID: hmYX5Yce
                                                AND
                                                Exp (4)
                                                  Exp (4)
                                                    INT: 26893
                                                  RELOP
                                                  Exp (4)
                                                    INT: 170953
                                              AND
                                              Exp (4)
                                                INT: 561133
                                            OR
                                            Exp (4)
                                              FLOAT: 279.799988
                                          OR
                                          Exp (4)
                                            Exp (4)
                                              Exp (4)
                                                Exp (4)
                                                  INT: 773
                                                STAR
                                                Exp (4)
                                                  Exp (4)
                                                    FLOAT: 834.276001
                                                  DOT
                                                  ID: sUpsPeOdN
                                              STAR
                                              Exp (4)
                                                FLOAT: 130.130005
                                            DIV
                                            Exp (4)
                                              LP
                                              Exp (4)
                                                Exp (4)
                                                  ID: h9FmcczyZf
                                                STAR
                                                Exp (4)
                                                  Exp (4)
                                                    Exp (4)
                                                      INT: 19141
                                                    DOT
                                                    ID: OB5F2
                                                  DOT
                                                  ID: MqXTklUkdkm
                                              RP
                                        OR
                                        Exp (4)
                                          ID: R1yWal
                                      OR
                                      Exp (4)
                                        MINUS
                                        Exp (4)
                                          FLOAT: 301.102203
                                    ASSIGNOP
                                    Exp (4)
                                      Exp (4)
                                        Exp (4)
                                          Exp (4)
                                            ID: XRSSlugn
                                            LP
                                            Args (4)
                                              Exp (4)
                                                ID: Ot9cOjKTA
                                              COMMA
                                              Args (4)
                                                Exp (4)
                                                  INT: 67
                                            RP
                                          AND
                                          Exp (4)
                                            INT: 0
                                        OR
                                        Exp (4)
                                          Exp (4)
                                            MINUS
                                            Exp (4)
                                              INT: 2134
                                          AND
                                          Exp (4)
                                            INT: 189
                                      OR
                                      Exp (4)
                                        Exp (4)
                                          NOT
                                          Exp (4)
                                            INT: 480985
                                        DIV
                                        Exp (4)
                                          ID: uDn
                                          LP
                                          Args (4)
                                            Exp (4)
                                              Exp (4)
                                                FLOAT: 65271930880.000000
                                              OR
                                              Exp (4)
                                                FLOAT: 79.860001
                                            COMMA
                                            Args (4)
                                              Exp (4)
                                                Exp (4)
                                                  Exp (4)
                                                    Exp (4)
                                                      Exp (4)
                                                        Exp (4)
                                                          Exp (4)
                                                            FLOAT: 2.235000
                                                          STAR
                                                          Exp (4)
                                                            ID: eofrN4
                                                            LP
                                                            RP
                                                        DIV
                                                        Exp (4)
                                                          ID: a1_ZYF1eXu
                                                      RELOP
                                                      Exp (4)
                                                        ID: TjhP_na3
                                                        LP
                                                        RP
                                                    RELOP
                                                    Exp (4)
                                                      ID: iIkFME0
                                                      LP
                                                      RP
                                                  OR
                                                  Exp (4)
                                                    Exp (4)
                                                      INT: 3468
                                                    DOT
                                                    ID: M8yRScvm
                                                ASSIGNOP
                                                Exp (4)
                                                  Exp (4)
                                                    Exp (4)
                                                      INT: 61000
                                                    PLUS
                                                    Exp (4)
                                                      INT: 1304
                                                  RELOP
                                                  Exp (4)
                                                    Exp (4)
                                                      Exp (4)
                                                        NOT
                                                        Exp (4)
                                                          ID: IolK5X
                                                          LP
                                                          RP
                                                      DIV
                                                      Exp (4)
                                                        ID: pHIponW
                                                    STAR
                                                    Exp (4)
                                                      FLOAT: 0.788000
                                          RP
                              COMMA
                              Args (4)
                                Exp (4)
                                  Exp (4)
                                    MINUS
                                    Exp (4)
                                      INT: 979
                                  OR
                                  Exp (4)
                                    NOT
                                    Exp (4)
                                      Exp (4)
                                        INT: 6382
                                      LB
                                      Exp (4)
                                        Exp (4)
                                          Exp (4)
                                            MINUS
                                            Exp (4)
                                              NOT
                                              Exp (4)
                                                NOT
                                                Exp (4)
                                                  NOT
                                                  Exp (4)
                                                    ID: OBMfYSX
                                                    LP
                                                    RP
                                          RELOP
                                          Exp (4)
                                            ID: sg6yTxmB
                                        AND
                                        Exp (4)
                                          Exp (4)
                                            NOT
                                            Exp (4)
                                              INT: 1158
                                          RELOP
                                          Exp (4)
                                            Exp (4)
                                              ID: O6RzfjwHp
                                              LP
                                              RP
                                            PLUS
                                            Exp (4)
                                              ID: npUWxdGj0
                                      RB
                            RP
                      AND
                      Exp (4)
                        Exp (4)
                          Exp (4)
                            FLOAT: 4.964500
                          DIV
                          Exp (4)
                            FLOAT: 8.900000
                        MINUS
                        Exp (4)
                          INT: 2491
                  RP
              SEMI
            StmtList (5)
              Stmt (5)
                Exp (5)
                  Exp (5)
                    NOT
                    Exp (5)
                      FLOAT: 0.000038
                  AND
                  Exp (5)
                    ID: EDczaDe
                SEMI
              StmtList (6)
                Stmt (6)
                  IF
                  LP
                  Exp (6)
                    Exp (6)
                      ID: Ysyy6vB
                    OR
                    Exp (6)
                      Exp (6)
                        INT: 1318432
                      RELOP
                      Exp (6)
                        Exp (6)
                          Exp (6)
                            ID: O39q3JpQz
                            LP
                            Args (6)
                              Exp (6)
                                Exp (6)
                                  INT: 2248
                                STAR
                                Exp (6)
                                  INT: 2080
                            RP
                          DOT
                          ID: JJoPm2
                        PLUS
                        Exp (6)
                          FLOAT: 389.000000
                  RP
                  Stmt (6)
                    Exp (6)
                      Exp (6)
                        FLOAT: 6.415800
                      MINUS
                      Exp (6)
                        ID: eMg1t8
                    SEMI
        RC
    ExtDefList (8)
      ExtDef (8)
        Specifier (8)
          StructSpecifier (8)
            STRUCT
            LC
            DefList (9)
              Def (9)
                Specifier (9)
                  TYPE: int
                DecList (9)
                  Dec (9)
                    VarDec (9)
                      VarDec (9)
                        ID: La7LG2_6_Aw
                      LB
                      INT: 62973
                      RB
                  COMMA
                  DecList (9)
                    Dec (9)
                      VarDec (9)
                        ID: I4vL87mj3
                      ASSIGNOP
                      Exp (9)
                        Exp (9)
                          INT: 622277
                        ASSIGNOP
                        Exp (9)
                          ID: JbkjiC4b
                          LP
                          Args (9)
                            Exp (9)
                              ID: ylpyb_
                              LP
                              Args (9)
                                Exp (9)
                                  INT: 110
                              RP
                          RP
                    COMMA
                    DecList (9)
                      Dec (9)
                        VarDec (9)
                          ID: VXJeN
                      COMMA
                      DecList (9)
                        Dec (9)
                          VarDec (9)
                            ID: m5y1Q1Y9
                          ASSIGNOP
                          Exp (9)
                            FLOAT: 40.279999
                SEMI
              DefList (10)
                Def (10)
                  Specifier (10)
                    TYPE: float
                  DecList (10)
                    Dec (10)
                      VarDec (10)
                        VarDec (10)
                          VarDec (10)
                            ID: SQEup9
                          LB
                          INT: 379
                          RB
                        LB
                        INT: 302
                        RB
                    COMMA
                    DecList (10)
                      Dec (10)
                        VarDec (10)
                          ID: Qtixt0HLG3_
                  SEMI
                DefList (11)
                  Def (11)
                    Specifier (11)
                      TYPE: float
                    DecList (11)
                      Dec (11)
                        VarDec (11)
                          VarDec (11)
                            VarDec (11)
                              ID: dznEm
                            LB
                            INT: 9237
                            RB
                          LB
                          INT: 1036
                          RB
                    SEMI
                  DefList (12)
                    Def (12)
                      Specifier (12)
                        TYPE: int
                      DecList (12)
                        Dec (12)
                          VarDec (12)
                            ID: sKezEt
                          ASSIGNOP
                          Exp (12)
                            Exp (12)
                              ID: k8TeL3
                              LP
                              Args (12)
                                Exp (12)
                                  ID: iO2Jnx
                                  LP
                                  RP
                                COMMA
                                Args (12)
                                  Exp (12)
                                    ID: QI6_M_
                                  COMMA
                                  Args (12)
                                    Exp (12)
                                      FLOAT: 0.001400
                                    COMMA
                                    Args (12)
                                      Exp (12)
                                        Exp (12)
                                          Exp (12)
                                            ID: IPnDn3dle
                                          LB
                                          Exp (12)
                                            ID: DSEZ4Ie
                                          RB
                                        RELOP
                                        Exp (12)
                                          LP
                                          Exp (12)
                                            Exp (12)
                                              FLOAT: 51.790001
                                            LB
                                            Exp (12)
                                              Exp (12)
                                                ID: EcMYvqls
                                              OR
                                              Exp (12)
                                                Exp (12)
                                                  NOT
                                                  Exp (12)
                                                    INT: 80988
                                                DIV
                                                Exp (12)
                                                  Exp (12)
                                                    Exp (12)
                                                      INT: 75968
                                                    DOT
                                                    ID: aI0lfWN
                                                  DOT
                                                  ID: GuuKpZH
                                            RB
                                          RP
                                      COMMA
                                      Args (12)
                                        Exp (12)
                                          NOT
                                          Exp (12)
                                            FLOAT: 534200.000000
                                        COMMA
                                        Args (12)
                                          Exp (12)
                                            Exp (12)
                                              Exp (12)
                                                LP
                                                Exp (12)
                                                  Exp (12)
                                                    FLOAT: 10.900000
                                                  ASSIGNOP
                                                  Exp (12)
                                                    ID: u2kUMN
                                                RP
                                              DIV
                                              Exp (12)
                                                FLOAT: 316560.000000
                                            OR
                                            Exp (12)
                                              Exp (12)
                                                Exp (12)
                                                  Exp (12)
                                                    NOT
                                                    Exp (12)
                                                      MINUS
                                                      Exp (12)
                                                        ID: ikfM
                                                        LP
                                                        Args (12)
                                                          Exp (12)
                                                            Exp (12)
                                                              ID: W70ir
                                                            DIV
                                                            Exp (12)
                                                              ID: jDPylQF
                                                              LP
                                                              Args (12)
                                                                Exp (12)
                                                                  Exp (12)
                                                                    FLOAT: 919.270020
                                                                  STAR
                                                                  Exp (12)
                                                                    ID: hm9Ulz
                                                                    LP
                                                                    Args (12)
                                                                      Exp (12)
                                                                        Exp (12)
                                                                          ID: XdxGuieEPu
                                                                        DOT
                                                                        ID: enPvJK
                                                                      COMMA
                                                                      Args (12)
                                                                        Exp (12)
                                                                          Exp (12)
                                                                            FLOAT: 413030907904.000000
                                                                          PLUS
                                                                          Exp (12)
                                                                            ID: PVna
                                                                    RP
                                                              RP
                                                          COMMA
                                                          Args (12)
                                                            Exp (12)
                                                              FLOAT: 12312000.000000
                                                            COMMA
                                                            Args (12)
                                                              Exp (12)
                                                                ID: YkiveSwM
                                                              COMMA
                                                              Args (12)
                                                                Exp (12)
                                                                  FLOAT: 378093.000000
                                                        RP
                                                  MINUS
                                                  Exp (12)
                                                    Exp (12)
                                                      Exp (12)
                                                        Exp (12)
                                                          NOT
                                                          Exp (12)
                                                            INT: 1711
                                                        STAR
                                                        Exp (12)
                                                          FLOAT: 4016.639893
                                                      STAR
                                                      Exp (12)
                                                        INT: 183
                                                    STAR
                                                    Exp (12)
                                                      FLOAT: 1.700000
                                                RELOP
                                                Exp (12)
                                                  Exp (12)
                                                    INT: 1036
                                                  DOT
                                                  ID: kPQDdsg
                                              AND
                                              Exp (12)
                                                FLOAT: 4.541000
                              RP
                            AND
                            Exp (12)
                              Exp (12)
                                FLOAT: 7245.319824
                              PLUS
                              Exp (12)
                                INT: 638
                      SEMI
            RC
        SEMI
      ExtDefList (15)
        ExtDef (15)
          Specifier (15)
            TYPE: float
          ExtDecList (15)
            VarDec (15)
              ID: SeNi2Z
            COMMA
            ExtDecList (15)
              VarDec (15)
                VarDec (15)
                  VarDec (15)
                    ID: V3pnGzkQEKr
                  LB
                  INT: 11
                  RB
                LB
                INT: 489
                RB
              COMMA
              ExtDecList (15)
                VarDec (15)
                  VarDec (15)
                    ID: U4kFnX5QmE
                  LB
                  INT: 20686
                  RB
                COMMA
                ExtDecList (15)
                  VarDec (15)
                    VarDec (15)
                      ID: FDV8umb
                    LB
                    INT: 9806
                    RB
                  COMMA
                  ExtDecList (15)
                    VarDec (15)
                      ID: Z8Tbar
                    COMMA
                    ExtDecList (15)
                      VarDec (15)
                        ID: Skm6YH
                      COMMA
                      ExtDecList (15)
                        VarDec (15)
                          ID: l2yCIRu
          SEMI
